packages/room/channel/channel.js

```diff
@@ -122,26 +123,55 @@ export const createChannel = ({ api, event, peer, streams }) => {
     _onError = async () => {
       const errorTime = Date.now()
 
-      if (this._roomId && this._clientId) {
-        const response = await this._api.getClient(this._roomId, this._clientId)
-
-        if (response.code === 404) {
+      const onError = async () => {
+        if (errorTime - Date.now() > 5000) {
           this.disconnect()
           this._event.emit(RoomEvent.CHANNEL_CLOSED, {
-            reason: REASONS.NOT_FOUND,
+            reason: REASONS.TIMEOUT,
           })
-          return
         }
-
-        // Reconnect
-        if (errorTime - this._startTime < 1000) {
+        if (!navigator.onLine) {
```

Comment:
@GagahGK , Why did you remove this `return` after timeout? From this logic, when the channel is closed because of timeout, the logic will continue to check the `!navigator.onLine` part. Is this intentional?

Reply:
ah I didn't realize the return was not written, thankyou for checking up on this changes